pennylane/templates/subroutines/double_excitation_unitary.py

```diff
@@ -395,8 +395,7 @@ def _layer8(weight, s, r, q, p, set_cnot_wires):
     RX(np.pi / 2, wires=p)
 
 
-@template
-def DoubleExcitationUnitary(weight, wires1=None, wires2=None):
+class DoubleExcitationUnitary(Operation):
```

Comment:
So is the idea that we would replace the decorator with class creation? I'm not sure I would fully agree with that., I like decorators! You also have to add `__init__` boilerplate with the class, and that just feels like extra typing.

I do however like the idea of having some custom ops/transforms as `Operation` subclasses. Having devices have their own control flow for these operations or just defaulting to decomposition is a very good idea. That way, we can always add cool new features without ever having to worry if it will break anything.

Reply:
I think it would be possible to keep the decorator UI just by reusing the infrastructure you have here.

Reply:
You mean having a decorator convert the template function into a class?

I don't think this will work easily - the `_preprocess` functions have a different signature and return structure for each template, and sometimes (as in `DoubleExcitation`) we have to do something with the wires...maybe for the simplest of templates we could support it. 

But when new data structures get added, the decorator has to automate more and more complex cases, so maybe better to not even start with it?